Issue/sortshift

pandyama opened the pull request:
- Modified database definition to include a new column for DayOfTheYear
- Used a sorting algorithm to sort the shifts from latest to oldest day
- Created a Helper class that would hold standalone functions so that the adapter class doesnt get too big

pandyama committed:
Added changes for new DayOfYear Column in db

pandyama committed:
Added sorting algorithm to sort the shifts by DayOfYear

pandyama committed:
made new Helper class for independent functions & put dayOfTheYear func there

oyiza commented on app/src/main/java/com/example/oddhours/ui/home/HomeAdapter.kt:
```diff
@@ -59,6 +59,9 @@ class HomeAdapter(private var jobList: List<JobModel>, val context: Context, pri
     val month = c.get(Calendar.MONTH)
     val day = c.get(Calendar.DAY_OF_MONTH)
 
+    val daysInMonth = intArrayOf(31,28,31,30,31,30,31,31,30,31,30,31)
```
have you considered leap year here? i.e. february will have 29 days instead of 28

pandyama replied:
I definitely missed considering that. Will work on that!!

oyiza commented on app/src/main/java/com/example/oddhours/data/repository/JobRepository.kt:
```diff
@@ -91,7 +102,8 @@ class JobRepository() {
     fun getShiftsForUIList(): MutableList<ShiftsListModel> {
-        val jobModelList = buildJobList() // TODO: maybe at this point we don't need to call buildJobList()? just use the jobModelList?
+        val jobModelList =
+            buildJobList() // TODO: maybe at this point we don't need to call buildJobList()? just use the jobModelList?
```
i dont mind moving code to new line to improve readability like we're doing for long function parameters, but i'm not sure if there's a need here?

pandyama replied:
sorry i am not following what you mean here?

oyiza replied:
it looks like the only change here is moving the rest of the code after the '=' sign to a new line?

oyiza commented on app/src/main/java/com/example/oddhours/data/repository/JobRepository.kt:
```diff
@@ -102,15 +114,34 @@ class JobRepository() {
         for (job in jobModelList) {
             shiftsFromJobId = dbShifts.getShiftsForJobID(job.jobID)
+
+            var swap = true
```
i was hoping we'd store the `shiftDayOfYear` variable and use it to sort in the ui simply by using a list sort method already available... that way we won't need to write our own function to sort it
 example: in `ParentAdapter` (~line 21) we're attaching the shifts list in reverse by using a list method. i was hoping we could utilize another list method to solve this, would look something like this:

`adapter = ChildAdapter(item.shifts.sortedByDescending { ShiftsModel -> ShiftsModel.shiftDayOfYear }, context, navController)`

i haven't tested this in code so syntax might be wrong but this site should help: https://www.codevscolor.com/kotlin-5-ways-sort-list-ascending-descending (the sortedByDescending() method)

pandyama replied:
i tried this out on an online kotlin compiler, it seem to be working. I'll work on implementing it using that built in method.

oyiza commented on app/src/main/java/com/example/oddhours/ui/home/HomeAdapter.kt:
```diff
@@ -113,6 +116,9 @@ class HomeAdapter(private var jobList: List<JobModel>, val context: Context, pri
                     // Log.d(TAG, "startDate: year: $year, month: ${monthOfYear}, day: $dayOfMonth")
                     // Log.d(TAG, "$startDate")
+                    dayOfYear = 0
+                    calculateDayOfTheYear(monthOfYear, dayOfMonth)
```
maybe this can use the same method from the helper class? so we don't duplicate code / effort here too

pandyama replied:
yaa for sure, will modify that.

oyiza commented on app/src/main/java/com/example/oddhours/utils/Helper.kt:
```diff
@@ -0,0 +1,20 @@
+package com.example.oddhours.utils
+
+class Helper {
```
maybe consider giving the helper class a more descriptive name? usually we'd go for something like `JobHelper` or `ShiftHelper` or `RepositoryHelper` ... totally up to you. since the codebase isn't complicated i'm also fine having just one helper class for all utility functions

pandyama replied:
ShiftHelper works. I just called it Helper for now, cuz I wasn't sure how much it'd be growing.

oyiza replied:
ya man no worries. the code base isn't super complicated at the moment so either separating it or leaving it all in one helper class is fine

pandyama committed:
Removed CalculateDayOfYear from HomeAdapter & fixed 1 line declaration in JobRepository

pandyama committed:
Removed sorting algorithm and replaced it with built in sortedBy method

pandyama committed:
Added Leap Year check in Helper class

pandyama commented:
- Removed CalculateDayOfYear from HomeAdapter and using Helper class now
- Removed Sorting Algorithm and using built in method to sort
- Added a check for Leap Year in Helper Class

oyiza reviewed (approved):
looks good man. thanks for working on this i've kind of been passive on this

pandyama commented:
> looks good man. thanks for working on this i've kind of been passive on this

ya bro anytime!!
